src/server.c

```diff
@@ -5637,20 +5737,24 @@ int prepareForShutdown(int flags) {
         TerminateModuleForkChild(server.child_pid,0);
     }
 
-    if (server.aof_state != AOF_OFF) {
-        /* Kill the AOF saving child as the AOF we already have may be longer
-         * but contains the full dataset anyway. */
-        if (server.child_type == CHILD_TYPE_AOF) {
-            /* If we have AOF enabled but haven't written the AOF yet, don't
-             * shutdown or else the dataset will be lost. */
-            if (server.aof_state == AOF_WAIT_REWRITE) {
+    /* Kill the AOF saving child as the AOF we already have may be longer
+     * but contains the full dataset anyway. */
+    if (server.child_type == CHILD_TYPE_AOF) {
+        /* If we have AOF enabled but haven't written the AOF yet, don't
+         * shutdown or else the dataset will be lost. */
+        if (server.aof_state == AOF_WAIT_REWRITE) {
+            if (force) {
+                serverLog(LL_WARNING, "Writing initial AOF. Exit anyway.");
+            } else {
                 serverLog(LL_WARNING, "Writing initial AOF, can't exit.");
-                return C_ERR;
+                goto error;
             }
-            serverLog(LL_WARNING,
-                "There is a child rewriting the AOF. Killing it!");
-            killAppendOnlyChild();
         }
+        serverLog(LL_WARNING,
+                  "There is a child rewriting the AOF. Killing it!");
+        killAppendOnlyChild();
+    }
+    if (server.aof_state != AOF_OFF) {
         /* Append only file: flush buffers and fsync() the AOF at exit */
         serverLog(LL_NOTICE,"Calling fsync() on the AOF file.");
         flushAppendOnlyFile(1);
```

Comment:
technically speaking, in current code, i guess we can change this to `server.aof_state == AOF_ON`, since in WAIT_REWRITE there's no "current" file to flush  to.

however, soon, in #9788 this is gonna change, and in case the state is WAIT_REWRITE, we'll want to delete the new part.

@chenyang8094 please confirm what i state above, and let's change that code in this PR to be:
1. safe / correct in current unstable 
2. make sure it'll cause a merge conflict in your PR so that you'll handle that case there too.

Reply:
@oranagra do you want me to change to `server.aof_state == AOF_ON` in this PR?

I don't think #9788 affects this.

Reply:
I think it's the right thing to do (change that condition), specifically if this PR is the last one, and then we publish a release.
but considering that we're going to change things again soon, and considering that this PR didn't break anything (the condition was wrong to before, and it has no actual implications.
let's keep the current code in this PR.